### ISSUE TITLE
Supabase Adapter not supporting sessions

### ISSUE THREAD

**Issue by denoni** [nextauthjs/next-auth#11635]
### Environment

```
  System:
    OS: macOS 14.5
    CPU: (8) arm64 Apple M1
    Memory: 45.59 MB / 8.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 21.7.1 - /opt/homebrew/bin/node
    Yarn: 1.22.22 - /opt/homebrew/bin/yarn
    npm: 10.5.0 - /opt/homebrew/bin/npm
    bun: 1.0.31 - /opt/homebrew/bin/bun
  Browsers:
    Chrome: 127.0.6533.120
    Safari: 17.5
  npmPackages:
    @auth/supabase-adapter: ^1.4.1 => 1.4.1 
    next: 14.2.3 => 14.2.3 
    next-auth: ^5.0.0-beta.19 => 5.0.0-beta.19 
    react: ^18 => 18.3.1 
```

### Reproduction URL

https://github.com/denoni/next-auth/tree/main

### Describe the issue

Using the doc as provided in [here ](https://authjs.dev/getting-started/adapters/supabase)was causing me the following problem: `[auth][cause]: JWEInvalid: Invalid Compact JWE`, what solved the problem was specifying the `strategy` as `jwt`, however that caused another issue related to `user.id` and `user.email` being used(they are `null` when `strategy: jwt`) instead of `token.id` and `token.email`

### How to reproduce

Basically just following step by step of what this guide suggests will lead to the error: https://authjs.dev/getting-started/adapters/supabase

### Expected behavior

Supabase should be able to support sessions and not just jwt.
As of: https://github.com/nextauthjs/next-auth/pull/11418#issuecomment-2278043731 - @balazsorban44 

A way to reproduce the correct behavior and see that the problems only occurs with the default `strategy` is to follow the [doc](https://authjs.dev/getting-started/adapters/supabase) but force `session: { strategy: "jwt" }` and change `async session({ session, token })`  instead of `async session({ session, user })`

**Comment by balazsorban44** [comment]
Forking our repo is not a reproduction...
